src/AlertsAggregator.py

```diff
@@ -0,0 +1,222 @@
+                filtered_df = self.df[self.df['data'].isin(user_settlement_lst)]
+                alert_amount = filtered_df.shape[0]
+                return {"message" : f"The total amount of alerts in {settlement} is: {alert_amount}"}
+            elif is_settlement_exist:
+                return {"message" : f"There were no alarms in the settlement: {settlement}"}
+            else:
+                raise ValueError("You selected a Settlement that does not exist.")
+        
+        except requests.exceptions.RequestException as req_err:
+            raise requests.exceptions.RequestException(f"Error occurred: {req_err}")
+        
+        except WrongSettlementException as settl_exc:
+            raise WrongSettlementException(settl_exc)
+
+        except ValueError as val_err:
+            raise ValueError(val_err)
+
+        
+    # This function receives df and count the total amount of alerts in the requested date range
+    def total_alerts_count(self):
+        total_alerts = self.df.shape[0]
+        return {"message" : f"The total amount of alerts in our country is: {total_alerts}"}
+    
+    def create_15_minute_interval_time_lst(self, start_time : time, end_time :time):
+        time_list = []
+        current_time = time(start_time.hour, 0)
+        print(current_time)
+        while current_time <= end_time:
+            time_list.append(current_time.strftime('%H:%M'))
+            current_time = (datetime.combine(datetime.today(), current_time) + timedelta(minutes=15)).time()
+
+        return time_list
+    
+    # This function receives city, and range of time and returns a df distribution of the alarms in quarters of an hour 0(0-15) 1(15-30) 2(30-45) 3(45-0)
+    def create_adjusted_time_column(self, settlement :str, start_time : time, end_time :time):
+        try:    
+            settlement_lst= self.create_user_settl_lst(settlement)
+            is_settlement_exist = self.does_the_settlement_exist(settlement)
+            # Extract the hour
+            start_hour = start_time.hour
+            end_hour = end_time.hour
+
+            if settlement_lst != []:
+                filtered_df = self.df[self.df['data'].isin(settlement_lst)]
+                filtered_df['time'] = pd.to_datetime(filtered_df['time']) 
+                # Filter the deteframe by the start and end time of the user
+                filtered_df = filtered_df[(filtered_df['time'].dt.hour >= start_hour) & (filtered_df['time'].dt.hour < end_hour)]
+                
+                # Create new column which represent the minutes by the nearest 15-minute interval
+                filtered_df['adjusted_time'] = filtered_df['time'].dt.strftime('%H:%M')
+                filtered_df['adjusted_time'] = pd.to_datetime(filtered_df['adjusted_time']).dt.round('15min').dt.strftime('%H:%M')
+                print(filtered_df)
+                
+                time_list = self.create_15_minute_interval_time_lst(start_time, end_time)
```

Comment:
add documentation inside the function to explain what you are doing here.
it is not clear at all.